src/modeler/AutoRest.Swagger/Validation/Extensions/LongRunningResponseValidationPost.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Validation;
+using AutoRest.Core.Properties;
+using System;
+using System.Collections.Generic;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class LongRunningResponseValidationPost : LongRunningExtensionRule
+    {
+        /// <summary>
+        /// The template message for this Rule.
+        /// </summary>
+        public override string MessageTemplate => Resources.LongRunningResponsePostNotValid;
+
+        /// <summary>
+        /// An x-ms-long-running-operation extension passes this rule if the operation that this extension has a valid response defined.
+        /// </summary>
+        /// <param name="longRunning">long running extension object.</param>
+        /// <param name="context">Rule context.</param>
+        /// <param name="formatParameters">list of parameters for message formatter.</param>
+        /// <returns><c>true</c> if operation has valid response code modeled, otherwise <c>false</c>.</returns>
+        /// <remarks>This rule corresponds to M2005.</remarks>
+        public override bool IsValid(object longRunning, RuleContext context, out object[] formatParameters)
+        {
+            return IsPostValid(context, out formatParameters);
+        }
+
+        /// <summary>
+        /// Validate that Put or Patch long running operations must have valid terminal status code 200 or 201.
+        /// </summary>
+        /// <param name="context">Rule context.</param>
+        /// <param name="formatParameters">list of parameters for message formatter.</param>
+        /// <returns><c>true</c> if at least one valid terminal status code availabe, otherwise <c>false</c>.</returns>
+        private static bool IsPostValid(RuleContext context, out object[] formatParameters)
```

Comment:
Yes I can but personally feels like not spending too much time on this because that'd lead me into Func delegator into C# and seems like overkill. Let me know if you guys think otherwise.